security/advancedtls/advancedtls.go

```diff
@@ -204,6 +213,15 @@ func (o *ServerOptions) config() (*tls.Config, error) {
 		return nil, fmt.Errorf(
 			"server needs to provide custom verification mechanism if choose to skip default verification, but require client certificate(s)")
 	}
+	clientCAs := o.RootCACerts
+	if o.VType != SkipVerification && o.RootCACerts == nil && o.GetRootCAs == nil {
```

Comment:
For server side, it is not always true that server needs trust certs. You also want to check `o.RequireClientCert == true`